source/base/data_out_base.cc

```diff
@@ -7786,6 +7786,11 @@ DataOutInterface<dim, spacedim>::write_vtu_in_parallel(
       }
   }
 
+  // Make sure we sync to disk. This ensures that the change in file
+  // size caused by the write_at() for the footer is visible for all
+  // ranks. Otherwise, the footer is sometimes lost.
+  ierr = MPI_File_sync(fh);
+  AssertThrowMPI(ierr);
 
   ierr = MPI_File_close(&fh);
```

Comment:
Do you need this because otherwise, some processes already close the file while others are still writing?

It is true that `MPI_File_close` is documented as implying `MPI_File_sync`, which would make this a bug. It might be nice to reflect these considerations in the comment, possibly a follow-up patch if @tamiko decides to merge this patch as soon as it tests correctly. Otherwise, someone will remove the sync call again sometime.

Reply:
This might well be a bug in the openmpi-4.1 library that is deployed on the cluster - if you want I can investigate whether this is an issue and has been fixed.

In either case the sync doesn't hurt (at least if I am reading the documentation right). So I would like to see this pull request merged :-)